magento/magento2
# Fix delay initialization options for customized JQuery UI menu widget

### Description
Documentation: http://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_menu.html

describes the ability to set the delay on the JQuery widget opening/closing with an option 'delay'. 

However, in the code, no such option appears and two undocumented options that aren't used do appear. 

This PR cleans up the unused variables and passes the delay parameter into JQuery menu widget so it is used. 


### Manual testing scenarios
 - Navigate to the home page
 - Hover over main navigation tab. Observe 300ms for widget menu to open/expand
 - Edit vendor/magento/module-theme/view/frontend/templates/html/topmenu.phtml - change UL tag to be: 

`<ul data-mage-init='{"menu":{"delay": 800, "responsive":true, "expanded":true, "position":{"my":"left top","at":"left bottom"}}}'>`
- Hover over main navigation tab. Observe 800ms delay for menu to open/expand

### Contribution checklist
